xbmc/filesystem/Directory.cpp

```diff
@@ -252,8 +252,8 @@ bool CDirectory::GetDirectory(const CURL& url,
         }
       }
 
-      // cache the directory, if necessary
-      if (!(hints.flags & DIR_FLAG_BYPASS_CACHE))
+      // cache the directory, if it is not empty and no DIR_FLAG_BYPASS_CACHE flag is set
+      if (!(hints.flags & DIR_FLAG_BYPASS_CACHE) && (items.Size() > 0))
```

Comment:
avoid use resources to calculate the size, just check for empty
```suggestion
      if (!(hints.flags & DIR_FLAG_BYPASS_CACHE) && !items.IsEmpty())
```